ec/ecresource/deploymentresource/update.go

```diff
@@ -83,7 +92,11 @@ func (r *Resource) Update(ctx context.Context, req resource.UpdateRequest, resp
 	updatePrivateStateTrafficFilters(ctx, resp.Private, planRules)
 	resp.Diagnostics.Append(v2.HandleRemoteClusters(ctx, r.client, plan.Id.ValueString(), plan.Elasticsearch)...)
 
-	deployment, diags := r.read(ctx, plan.Id.ValueString(), &state, &plan, res.Resources, planRules)
+	var newMigrateTemplateRequest *deployments.MigrateDeploymentTemplateOK
+	deployment, diags := r.read(ctx, plan.Id.ValueString(), &state, &plan, res.Resources, planRules, newMigrateTemplateRequest)
```

Comment:
If DT has not been changed in plan there is no need to make another migrate API call, I guess.

Reply:
It shouldn't make another request because now we only perform the request inside `read` [if the template has changed](https://github.com/elastic/terraform-provider-ec/pull/755/commits/a69026c9d9ed2acf9c0f3872779d6be75fb05bb6) (as you suggested above) 🙂

Reply:
Ahh, that's correct - my bad!